benches/threadpool_bench.rs

```diff
@@ -0,0 +1,113 @@
+use std::vec;
+
+use criterion::{black_box, criterion_group, criterion_main, Criterion};
+use reveaal::ProtobufServer::{
+    services::{
+        component::Rep, ecdar_backend_server::EcdarBackend, Component, ComponentsInfo, QueryRequest,
+    },
+    ConcreteEcdarBackend,
+};
+use tonic::Request;
+
+use criterion::async_executor::FuturesExecutor;
+use futures::stream::FuturesUnordered;
+use futures::StreamExt;
+
+mod flamegraph_profiler;
+use flamegraph_profiler::FlamegraphProfiler;
+
+static PATH: &str = "samples/json/EcdarUniversity";
+
+fn send_query_with_components(
+    id: String,
+    c: &mut Criterion,
```

Comment:
Maybe rename c to criterion

Reply:
idk, its seems to be the standard to just do it this way for criterion, this is also how all other benchmarks in reveaal are written.

Reply:
coolcool